### ISSUE TITLE
Building fails on Linux due to a missing fix in core-error

### ISSUE THREAD

**Issue by Schroedingers-Cat** [helgoboss/helgoboss-midi#8]
I wanted to try out this crate (it looks great btw), but `cargo build` is failing: 
```
error: failed to run custom build command for `core-error v0.0.0`

Caused by:
  process didn't exit successfully: `~/helgoboss-midi/target/debug/build/core-error-d2c9d08bd99bc69a/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'Rustc to give us its version', ~/.cargo/registry/src/github.com-1ecc6299db9ec823/core-error-0.0.0/build.rs:4:62
  stack backtrace:
     0: rust_begin_unwind
               at /rustc/1.59.0/library/std/src/panicking.rs:498:5
     1: core::panicking::panic_fmt
               at /rustc/1.59.0/library/core/src/panicking.rs:116:14
     2: core::panicking::panic_display
               at /rustc/1.59.0/library/core/src/panicking.rs:72:5
     3: core::panicking::panic_str
               at /rustc/1.59.0/library/core/src/panicking.rs:56:5
     4: core::option::expect_failed
               at /rustc/1.59.0/library/core/src/option.rs:1817:5
     5: core::option::Option<T>::expect
               at /rustc/1.59.0/library/core/src/option.rs:692:21
     6: build_script_build::main
               at ./build.rs:4:38
     7: core::ops::function::FnOnce::call_once
               at /rustc/1.59.0/library/core/src/ops/function.rs:227:5
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
Ironically, the dependency causing the error is called `core-error`.  

However, I can build that dependency just fine when manually cloning it from GitHub. It's being checked out with version `0.0.1-rc4`, so I tried changing the version in this repo's Cargo.toml to that version but it's still giving me the same error.  

I can only fix this directly referencing core-error's GitHub repo and not using its crate, like this:
```
[dependencies]
core-error = { git = "https://github.com/core-error/core-error.git", default-features = false }
```
It seems that core-error's last merge request fixes the issue I'm running into. However, it hasn't been published to crate.io, even though it's from late 2020: https://github.com/core-error/core-error/commit/1b00bcd1c1c900d035ec596ffc1b1b76e2b73098  

The problem is that the published version of core-error's build script uses version_check::triple() which returns Rust's version, channel and date and apparently the channel information isn't available on all platforms.  

For reference, the output of `rustc -vV` is:
```
rustc 1.59.0 (Arch Linux rust 1:1.59.0-1)
binary: rustc
commit-hash: unknown
commit-date: unknown
host: x86_64-unknown-linux-gnu
release: 1.59.0
LLVM version: 13.0.1
```

Would it be possible to use core-error's GitHub repo instead of its crate for the time being?

**Comment by helgoboss** [comment]
Thanks for looking into that! Interesting, this might also fix https://github.com/helgoboss/realearn/issues/538, which I could not reproduce in my setup. If I use `core-error`'s GitHub repo, I can't publish on crates.io, so this would only work if you use `helgoboss-midi` as GitHub repo dependency, too?

Wondering what's the best way to fix this long-term. Maybe to just not implement any error trait in no-std contexts. Then we can get rid of `core-error`.

**Comment by Schroedingers-Cat** [comment]
> Wondering what's the best way to fix this long-term. Maybe to just not implement any error trait in no-std contexts. Then we can get rid of core-error.  

That seems like the cleanest approach.  
Alternatively, asking the maintainer to publish the fix to crate.io or publishing a fork to crate.io should work, too.

**Comment by helgoboss** [comment]
This should be fixed with v0.4.0. I chose the first approach, just not implementing any error trait in no-std environments.
